### DISCUSSION TITLE
Allow MenuItems to open in new tab or window

### DISCUSSION THREAD

**Discussion by terrytrent** [zauberzeug/nicegui#960] [Ideas / Feature Requests]
I would like to be able to open MenuItems in a new tab.  Currently it only allows opening in the existing window using `lambda: ui.open('/page')`

This is my workaround:

```py
ui.menu_item('Page').props('onclick="window.open(\'/page\', \'_blank\');"')
```

**Comment by falkoschindler** [comment]
Great idea, @terrytrent!
So basically `ui.open` should accept a `new_tab` argument like `ui.link` does.

**Reply by thetableman** [reply]
@falkoschindler Is this functionality still a possibility? Allowing ui.open to open in a new browser tab would be a great addition.

**Reply by rodja** [reply]
No, this is not implemented yet. But it should be fairly easy to add. Would you like to create a pull request for this? That way you will be part of the NiceGUI contributors list.  See [CONTRIBUTING.md](https://github.com/zauberzeug/nicegui/blob/main/CONTRIBUTING.md#pull-requests) for a short explanation.

**Reply by thetableman** [reply]
@rodja Appreciate the confidence you have in my abilities but changing source code is well beyond my skill level. I've reviewed open.py but can't see how the href is executed, your python skill level is well beyond mine :)

**Comment by falkoschindler** [comment]
I just created PR #1243 implementing this feature. 🙂

**Comment by WolfgangFahl** [comment]
It works locally if i add "localhost" to browser permissions. Does not work as expected e.g. on http://nicesrt.bitplan.com/ so unfortunately the github link there also doesn't work: https://github.com/WolfgangFahl/nicesrt

**Reply by falkoschindler** [reply]
@WolfgangFahl I think it would be better to open a new discussion, ideally with a code example and more details about what you're trying to do and what "does not work".

**Reply by v479038280** [reply]
wow ,so cool . How can we use our own IP ? @falkoschindler  If we can make a tutorial about this

**Reply by falkoschindler** [reply]
@v479038280 Sorry, I don't understand what you mean.

**Reply by v479038280** [reply]
A. Tutorial about how to use our own ip address

**Comment by falkoschindler** [comment]
I'll close this feature request as completed in https://github.com/zauberzeug/nicegui/pull/1243.
